storage/series/series.go

```diff
@@ -370,14 +381,19 @@ func (s *dbSeries) mergeBlock(
 	// If we don't have an existing block just insert the new block.
 	existingBlock, ok := blocks.BlockAt(blockStart)
 	if !ok {
-		blocks.AddBlock(newBlock)
+		s.addBlock(newBlock)
 		return
 	}
 
 	// We are performing this in a lock, cannot wait for the existing
 	// block potentially to be retrieved from disk, lazily merge the stream.
 	newBlock.Merge(existingBlock)
-	blocks.AddBlock(newBlock)
+	s.addBlock(newBlock)
+}
+
+func (s *dbSeries) addBlock(b block.DatabaseBlock) {
```

Comment:
Maybe we should add suffix `...WithLock(...)` to methods `mergeBlock` and `addBlock` to make it clear.

Reply:
good call